x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/rest/SecurityRestFilter.java

```diff
@@ -100,6 +102,14 @@ private void handleException(String actionType, RestRequest request, RestChannel
                 @Override
                 protected boolean skipStackTrace() { return restStatus == RestStatus.UNAUTHORIZED; }
 
+                @Override
+                public Map<String, List<String>> filterHeaders(Map<String, List<String>> headers) {
+                    if (headers.containsKey("Warning")) {
```

Comment:
Do we want to remove _all_ warning headers on authentication failure?

Reply:
Yes, it is intentional (see Tim's comment above). If Authentication fails it does make sense to not reveal any additional information